packages/@react-spectrum/checkbox/src/Checkbox.tsx

```diff
@@ -27,6 +28,10 @@ import {useToggleState} from '@react-stately/toggle';
 
 function Checkbox(props: SpectrumCheckboxProps, ref: FocusableRef<HTMLLabelElement>) {
   let originalProps = props;
+  let inputRef = useRef<HTMLInputElement>(null);
+  let domRef = useFocusableRef(ref, inputRef);
+
+  [props, domRef] = useContextProps(props, domRef, CheckboxContext);
```

Comment:
RSP Checkbox need to be able to get the props from the RAC context setup by RAC Tree. Eventually be a moot point when we convert v3 to use RAC completely.